### PR TITLE
feat(cairo): update tree-sitter grammar and queries

### PR DESCRIPTION
**Pull request by LucasLvy**
Move from rust grammar to a proper cairo grammar. Both the grammar and the queries are based on the rust one

### CODE REVIEW

File: runtime/queries/cairo/highlights.scm [hunk 1]
```diff
@@ -1 +1,368 @@
...
+
+(parameter
+	pattern: (identifier) @variable.parameter)
+
+; -------
+; Keywords
+; -------
+
+((identifier) @keyword.control
+  (#match? @keyword.control "^yield$"))
```
Comment (the-mikedavis) [inline runtime/queries/cairo/highlights.scm:100]:
```suggestion
  (#eq? @keyword.control "yield"))
```

for an exact match you can use the `#eq?` predicate

**Comment by the-mikedavis** [comment]
Could you add a license to the grammar? That will help downstream packagers like Fedora

File: runtime/queries/cairo/highlights.scm (hunk 1 above)
Comment (LucasLvy) [reply runtime/queries/cairo/highlights.scm:100]: actually we don't even have  `yield` in cairo, copied a bit too much from rust

Comment (LucasLvy) [reply runtime/queries/cairo/highlights.scm]: done

**Comment by LucasLvy** [comment]
Added a license

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**